### DISCUSSION TITLE
Upgrade to latest version broken port 4000 access

### DISCUSSION THREAD

**Discussion by robbo1001** [teslamate-org/teslamate#2585] [Q&A]
Hi there. 

I installed Teslamate back in December on a raspberry pi using Docker. This morning I ran an update using the command:

docker-compose pull
docker-compose up -d

However, this seems to have broken the link between the Tesla and Teslamate. I can still access the Teslamate pages using port 3000, but they aren't updating anymore. Also, if I go to port 4000, I get "This site can’t be reached".

Can anyone give me any tips as to what I have done wrong, or if it is possible to revert to the previous version?

Thanks

**Comment by cwanja** [comment]
https://github.com/adriankumpf/teslamate/issues/2302

I assume you are running Buster? You likely need to upgrade to Bullseye or use some of the backwards compatibility fixes in that issue. But there's a handful of responses in there, including from Adrian.

**Reply by robbo1001** [reply]
Amazing - thanks. 

I had just started updating to Bullseye, and am now seeing if I can get it all working again. I should have done more research before posting. 

However, after now updating to Bullseye, I am getting some massive errors. My logs are extremely long with lots of reference to 

"teslamate_1 | waiting for postgres at database:5432"

I may have to completely start from scratch and delete the docker (not sure how to do that), and reinstall teslamate and reinstate a backup.

Any ideas?

**Reply by ktsenvolt** [reply]
Install Raspberry Pi OS Bullseye on Raspberry Pi (Illustrated guide)

    Download Raspberry Pi Imager on the website.
    Install it on your computer.
    Open Imager, and choose the Raspberry Pi OS version to install.
    Insert your SD card and write it.
    Start your Raspberry Pi with this card and complete the configuration.
Install docker and then teslamate.

**Reply by robbo1001** [reply]
Thanks all. 

I was keen to upgrade to Bullseye, rather than to do a clean install, because the same RPi also runs TVHeadend as my media server. 

After the update, I had to manually update the PostgreSQL database and reinstall the backup from Friday night. After a big (quite a lot) of fiddling, it is now all working again. 

Thanks for the help, and the great software!
